src/platform/mt793x/DnssdImpl.cpp

```diff
@@ -55,25 +188,56 @@ namespace Dnssd {
 #define SERVICE_DOMAIN ("local")
 
 MdnsContexts MdnsContexts::sInstance;
-static DNSServiceRef client = NULL;
-static TXTRecordRef PublishTxtRecord;
+static DNSServiceRef BrowseClient = NULL;
+static TaskHandle_t gResolveTask  = NULL;
+static EventGroupHandle_t gResolveTaskWakeEvent;
 
 void ChipDnssdMdnsLog(const char * level, const char * msg)
 {
     ChipLogProgress(ServiceProvisioning, "%s %s", StringOrNullMarker(level), StringOrNullMarker(msg));
 }
 
-/**
- * @brief     mDNS Daemon Task entry
- * @param[in] void *not_used:Not used
- * @return    None
- */
+static void OnRegister(DNSServiceRef sdRef, DNSServiceFlags flags, DNSServiceErrorType err, const char * name, const char * type,
+                       const char * domain, void * context)
+{
+    ChipLogDetail(Discovery, "Mdns: %s name: %s, type: %s, domain: %s, flags: %ld", __func__, name, type, domain, flags);
+
+    auto sdCtx = reinterpret_cast<RegisterContext *>(context);
+    sdCtx->Finalize(err);
+};
+
+CHIP_ERROR Register(void * context, DnssdPublishCallback callback, uint32_t interfaceId, const char * type, const char * name,
+                    uint16_t port, ScopedTXTRecord & record, Inet::IPAddressType addressType, const char * hostname)
+{
+    ChipLogDetail(Discovery, "Registering service %s on host %s with port %u and type: %s on interface id: %" PRIu32, name,
+                  hostname, port, type, interfaceId);
+
+    RegisterContext * sdCtx = nullptr;
+    if (CHIP_NO_ERROR == MdnsContexts::GetInstance().GetRegisterContextOfType(type, &sdCtx))
+    {
+        auto err = DNSServiceUpdateRecord(sdCtx->serviceRef, nullptr, kRegisterFlags, record.size(), record.data(), 0 /* ttl */);
+        VerifyOrReturnError(kDNSServiceErr_NoError == err, sdCtx->Finalize(err));
+        return CHIP_NO_ERROR;
+    }
+
+    sdCtx = chip::Platform::New<RegisterContext>(type, name, callback, context);
+    VerifyOrReturnError(nullptr != sdCtx, CHIP_ERROR_NO_MEMORY);
+
+    // sdCtx->mHostNameRegistrar.Init(hostname, addressType, interfaceId);
+
+    DNSServiceRef sdRef;
+    auto err = DNSServiceRegister(&sdRef, kRegisterFlags, interfaceId, name, type, kLocalDot, hostname, ntohs(port), record.size(),
```

Comment:
Shouldn't that be `htons(port)`?

Reply:
Yes `htons(port)` is right!
I think I can passed PICS test due to my test devices are all **litte-endian**, however if test device contains **big-endian**, I guess PICS test will fail.